livekit/src/room/mod.rs

```diff
@@ -328,6 +330,16 @@ impl Room {
         url: &str,
         token: &str,
         options: RoomOptions,
+    ) -> RoomResult<(Self, mpsc::UnboundedReceiver<RoomEvent>)> {
+        Self::connect_with_sdk(url, token, options, Some("rust"), Some(SDK_VERSION)).await
+    }
+
+    pub async fn connect_with_sdk(
```

Comment:
I think you can add those options inside `RoomOptions`. No need to create a new method. Let's keep everything inside the `connect` one?
Eitherway the fields will be public, there is no straightforward way to hide sdk & sdk_version to the users

Reply:
it felt extra weird inside `RoomOptions` to me... but yeah I can change it back

Reply:
@theomonnom I started on changing it back and found it kinda spiraled out pretty quick, RoomOptions is used a lot more places than need sdk and sdk_version, so there was a lot of unrelated surface area doing it that way. I decided to make a new options type called RoomAnalyticsOptions to contain them, to be passed alongside RoomOptions. And I decided to do the same thing for SignalClient.  It makes its purpose clearer. Maybe we could just change Room::connect  to take a new ConnectOptions instead of a RoomOptions, and ConnectOptions would include `roomOptions`, `sdk`, and `sdk_version`.

What do you think of this approach?

Reply:
the other problem with this is it becomes a breaking change in the rust RTC library, just so that we can add some internal stuff for FFI. IMO better to have a separate method that includes the sdk overrides and is marked for FFI use only?

Reply:
after further discussion with @theomonnom I've updated this to just include it in RoomOptions and SignalOptions. I chose to include it as a nested type rather than individual fields, as I anticipate adding at least one more field to it soon (for the `other_sdks` field to include the versions of associated sdks in use)